src/DotLiquid/Tokenizer.cs

```diff
@@ -0,0 +1,139 @@
+using DotLiquid.Exceptions;
+using DotLiquid.Util;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace DotLiquid
+{
+    internal static class Tokenizer
```

Comment:
Consider adding XML comments to the class and its method so the intend is clear. It will help code maintainers to distinguish intend from actual implementation while troubleshooting issues.

Reply:
Will consider. It is internal, so it is not required. Likewise the CharEnumerator.

Reply:
Yes for CharEnumerator it would make even more sense to add XML comments to answer questions just like I had in this review (i.e. why do we even have such class).

Reply:
Done for both.